components/sidebar/sidebar_service_unittest.cc

```diff
@@ -729,4 +877,175 @@ TEST_F(SidebarServiceTestWithPlaylist, GetDefaultPanelItem) {
+  SidebarServiceOrderingTest(const SidebarServiceOrderingTest&) = delete;
+  SidebarServiceOrderingTest& operator=(const SidebarServiceOrderingTest&) =
+      delete;
+  SidebarServiceOrderingTest(const SidebarServiceOrderingTest&&) = delete;
+  SidebarServiceOrderingTest& operator=(const SidebarServiceOrderingTest&&) =
+      delete;
+
+  ~SidebarServiceOrderingTest() override = default;
+
+  PrefService* GetPrefs() { return profile_->GetPrefs(); }
+
+  SidebarService* GetSidebarService() { return sidebar_service_; }
+
+  void SetUp() override {
+    testing::Test::SetUp();
+
+#if BUILDFLAG(ENABLE_AI_CHAT)
+    scoped_feature_list_.InitAndEnableFeature(ai_chat::features::kAIChat);
+#endif  // BUILDFLAG(ENABLE_AI_CHAT)
+
+    auto registry = base::MakeRefCounted<user_prefs::PrefRegistrySyncable>();
+    sidebar::SidebarServiceFactory::GetInstance();
+    RegisterUserProfilePrefs(registry.get());
+    registry->RegisterDictionaryPref("brave");
+    registry->RegisterDictionaryPref("brave.sidebar");
+
+    sync_preferences::PrefServiceMockFactory factory;
+    auto pref_service = factory.CreateSyncable(registry.get());
+    auto* pref_service_ptr = pref_service.get();
+
+    auto builder = TestingProfile::Builder();
+    builder.SetPrefService(std::move(pref_service));
+    profile_ = builder.Build();
+
+    ASSERT_EQ(pref_service_ptr, profile_->GetPrefs());
+  }
+
+  void InitSideBarService() {
+    sidebar_service_ =
+        sidebar::SidebarServiceFactory::GetForProfile(profile_.get());
+    sidebar_service_->AddObserver(&observer_);
+  }
+
+  bool ValidateBuiltInTypesOrdering(
+      const std::vector<SidebarItem::BuiltInItemType>& defined_order) {
+    size_t srv_items_index = 0, dbt_index = 0;
+    const auto default_btin_types_count = defined_order.size();
+    std::vector<SidebarItem> only_builtin_types;
+    base::ranges::copy_if(
+        sidebar_service_->items(), std::back_inserter(only_builtin_types),
+        [](const SidebarItem& item) {
+          return item.built_in_item_type != SidebarItem::BuiltInItemType::kNone;
+        });
+
+    const auto srv_items_count = only_builtin_types.size();
+    while (srv_items_index < srv_items_count &&
+           dbt_index < default_btin_types_count) {
+      if (only_builtin_types[srv_items_index].built_in_item_type ==
+          defined_order[dbt_index]) {
+        srv_items_index++;
+      }
+      dbt_index++;
+    }
+    return srv_items_index == srv_items_count;
+  }
+
+  void LoadFromPrefsTest(
+      base::Value::Dict sidebar_prefs,
+      const std::vector<SidebarItem::BuiltInItemType>& defined_order,
+      const size_t& expected_items_loaded) {
+    // base::Value::Dict
+    // sidebar(ParseTestJson(sidebar_all_builtin_visible_json));
+    GetPrefs()->Set(kSidebarItems,
+                    std::move(sidebar_prefs.Find("sidebar_items")->Clone()));
+    GetPrefs()->SetInteger(kSidebarShowOption, 0);
+    EXPECT_EQ(0, GetPrefs()->GetInteger(kSidebarShowOption));
```

Comment:
Maybe you want to set it from `sidebar_prefs` as you defined it in preset data?
but I think setting/checking show option here doesn't have much meaning.

Reply:
removed it